[mb2] Add the 'scrape' command. JB#59443

martyone opened the pull request.

vigejolla commented on sdk-setup/src/mb2:
```diff
@@ -2703,7 +2712,7 @@ init_patch_wrapper() (
         readarray -t untracked_added < <(join -v 2 \
             <(printf '%s' "$untracked_old") <(printf '%s' "$untracked_now"))
-        if [[ $untracked_added ]]; then
+        if [[ ${#untracked_added[*]} -gt 0 ]]; then
```
How is this related to -o nounset?

martyone replied:
The original code fails with empty array (or more exactly an array with item at index 0 unset).

vigejolla commented on sdk-setup/src/mb2:
```diff
@@ -531,9 +532,54 @@ COMMANDS
+        the '--no-vcs-apply' global option is used).
+
+        This command enforces certain patch formatting and file naming
+        conventions. It is not guarranteed (not desired) that it produces
```
extra 'r' in guaranteed

martyone replied:
I really hate this word :) fixed.

vigejolla reviewed (approved):
One question, which I would like answered, and one typo. Approving already; If you make no further changes besides fixing the typo, feel free to merge.

martyone commented:
Polished more

vigejolla reviewed (approved):
Still LGTM

martyone commented:
Extended it to remove the commits after creating patches from them, renamed the command to avoid any doubt on which command is meant (whether mb2's subcommand or git's subcommand), moved it to it's own domain so that it does not disturb the row of the closely related commands prep, build, qmake, make, package etc.

vigejolla reviewed (approved):
It just keeps getting better :)
